data/scripts/import_acs_data.py

```diff
@@ -0,0 +1,110 @@
+import os
+import sys
+import csv
+
+import yaml
+from census import Census
+
+# Relevant FIPS codes for geographic queries
+US_FIPS = '1'
+PA_FIPS = '42'
+PHILLY_FIPS = '101'
+
+
+class ACSWriter(object):
+    """
+    Thin wrapper around a csv.DictWriter to perform some common formatting
+    for ACS API responses.
+    """
+    def __init__(self, output, variables):
+        """
+        - output: A file-like object to write the CSV to
+        - variables: A dict of ACS variables parsed from a .yml config file
+        """
+        self.codes_to_names = {var['code']: var['name'] for var in variables}
+        fieldnames = ['fips'] + [var['name'] for var in variables]
+        self.writer = csv.DictWriter(output, fieldnames=fieldnames)
+
+    def writeheader(self):
+        self.writer.writeheader()
+
+    def write_acs_row(self, fips, acs_row):
+        row = {'fips': fips}
+        for var_code, var_stat in acs_row.items():
+            # Skip fields that don't match queried variables
+            if var_code not in self.codes_to_names.keys():
+                continue
+            # Process nulls
+            if var_stat == -666666666:
```

Comment:
I couldn't find any docs on this in the Census API. I was getting a few responses that looked like this from 2016 data, so I went ahead and treated them like nulls. I've since switched to 2017 data and I'm not sure whether the unusual responses persisted.

Reply:
this would be good to report upstream.

Reply:
Good call, I'll open an issue.

Reply:
Opened https://github.com/datamade/census/issues/72 for this.